src/annotator/sidebar.js

```diff
@@ -301,15 +301,16 @@ export class Sidebar {
     );
 
     // The listener will do nothing if the sidebar doesn't have a bucket bar
-    // (clean theme), but it is still actively listening.
-    guestRPC.on(
-      'anchorsChanged',
-      /** @param {AnchorPosition[]} positions  */
-      positions => {
-        // Currently, only one guest frame sends anchor positions to the bucket bar
-        this.bucketBar?.update(positions);
-      }
-    );
+    // (clean theme)
+    const bucketBar = this.bucketBar;
+    // Currently, we ignore `anchorsChanged` for all the guests except the first connected guest.
+    if (this._guestRPC.length === 0 && bucketBar) {
```

Comment:
There is a potential issue here if a new guest is added and then one is removed. For example if in an ebook the new chapter iframe was created and the old one was removed and the new guest was able to register itself before the old one unloaded. What we could do to avoid this is to always register the listener, but check if the guest is the first entry in the `this._guestRPC` array when the callback is executed.

Reply:
👍🏼